backend/apps/common/utils.py

```diff
@@ -194,13 +257,151 @@ def get_raw_deployment_objects(svc, component):
+    """
+    namespace = svc["metadata"]["namespace"]
+
+    objects = {
+        "predictor": None,
+        "servingRuntime": None,
+        "deployment": None,
+        "service": None,
+    }
+
+    try:
+        # In ModelMesh, the actual serving is handled by the
+        # ModelMesh controller
+        # Look for the predictor status information
+        if "status" in svc and "components" in svc["status"]:
+            if component in svc["status"]["components"]:
+                objects["predictor"] = svc["status"]["components"][component]
+                log.info(
+                    f"Found ModelMesh predictor status for " f"component {component}"
+                )
+
+        # Try to find the associated ServingRuntime
+        serving_runtime_name = None
+        if "spec" in svc and component in svc["spec"]:
+            runtime_ref = svc["spec"][component].get("runtime")
+            if runtime_ref:
+                serving_runtime_name = runtime_ref
+            else:
+                # Fallback: infer from model format
+                model_format = (
+                    svc["spec"][component]
+                    .get("model", {})
+                    .get("modelFormat", {})
+                    .get("name")
+                )
+                if model_format:
+                    serving_runtime_name = f"mlserver-{model_format.lower()}"
+
+        if serving_runtime_name:
+            try:
+                # Get ServingRuntime
+                serving_runtime = api.get_custom_rsrc(
+                    group="serving.kserve.io",
+                    version="v1alpha1",
+                    kind="servingruntimes",
+                    namespace=namespace,
+                    name=serving_runtime_name,
+                )
+                objects["servingRuntime"] = serving_runtime
+                log.info(
+                    f"Found ServingRuntime {serving_runtime_name} for "
+                    f"component {component}"
+                )
+            except Exception as e:
+                log.warning(
+                    f"Could not find ServingRuntime " f"{serving_runtime_name}: {e}"
+                )
+
+        # Find ModelMesh deployments using helper function
+        deployments = _find_modelmesh_resources(
+            namespace, versions.K8S_DEPLOYMENT, "deployment"
+        )
+        if deployments:
+            # In a typical ModelMesh setup, there should be one deployment per namespace
+            # that serves all models. If multiple deployments exist, we take the first one
+            # but log a warning for visibility.
+            if len(deployments) > 1:
+                deployment_names = [d["metadata"]["name"] for d in deployments]
+                log.warning(
+                    f"Multiple ModelMesh deployments found: {deployment_names}. Using first one: {deployment_names[0]}"
+                )
+
+            objects["deployment"] = deployments[0]
```

Comment:
I don't think it's safe to assume there's only one deployment per namespace. I think we could use some of the information above to match the deployment you need to the serving runtime name or something like that